plugin/gcp/cloudFunction.go

```diff
@@ -0,0 +1,55 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/puppetlabs/wash/plugin"
+	"google.golang.org/api/cloudfunctions/v1"
+)
+
+type cloudFunction struct {
+	plugin.EntryBase
+	service cloudFunctionsProjectService
+}
+
+func newCloudFunction(function *cloudfunctions.CloudFunction, service cloudFunctionsProjectService) *cloudFunction {
+	cf := &cloudFunction{
+		EntryBase: plugin.NewEntry(function.Name),
+		service:   service,
+	}
+	mtime, err := time.Parse(time.RFC3339, function.UpdateTime)
+	if err != nil {
+		panic(fmt.Sprintf("Timestamp for %v was not expected format RFC3339: %v", cf, function.UpdateTime))
+	}
+	cf.
+		Attributes().
+		SetMtime(mtime).
+		SetMeta(function)
+	return cf
+}
+
+func (cf *cloudFunction) List(ctx context.Context) ([]plugin.Entry, error) {
+	cfl, err := newCloudFunctionLog(ctx, cf.service, cf.Name())
+	if err != nil {
+		return nil, err
+	}
+	return []plugin.Entry{cfl}, nil
+}
+
+func (cf *cloudFunction) Delete(ctx context.Context) (bool, error) {
+	_, err := cf.service.Projects.Locations.Functions.Delete(cf.Name()).Context(ctx).Do()
+	return false, err
```

Comment:
Cloud functions have some shut down stuff that needs to happen before they're deleted, which is why this is false.